### PR TITLE
DAF-4255 - Inactive seek button while watching livestream in PiP mode

### PR DESCRIPTION
**Pull request by mvn-hienvu-hn**
## What was done
- Pass `isLiveStream` flag to native side in the cases of setting memory/file data source.
- Add `isLiveStream` flag when setting datasource for PiP plan limited video (to hide seek buttons on that video while it is playing on a livestream) [in the app repo](https://github.com/dwango-nfc/dwango-app-ch/pull/1715).
- Use `isLiveStream` flag to hide `seek` buttons on PiP while livestream is playing.

## Ticket

https://dw-ml-nfc.atlassian.net/browse/DAF-4255

## Spec

JP: https://docs.google.com/spreadsheets/d/1iXu4Vz9wYpT6fHE7oK_W4vyHP_-BGTvk7HTcETFJ_0I/edit#gid=2109148807&range=60:61

VN: https://docs.google.com/spreadsheets/d/1XLwXVbNo_r21YE8Ir4mDQcyAa9CbKDWA/edit#gid=876877031&range=60:61

## Screenshot
- LIVE/DVR

https://github.com/dwango-nfc/betterplayer/assets/100773699/90dace68-3584-4a35-bca7-fb9a964110ef

- Archiving Live stream:

https://github.com/dwango-nfc/betterplayer/assets/100773699/adcaf842-f319-4086-a198-e4f44360c0d6

### CODE REVIEW

File: ios/Classes/BetterPlayer.h [hunk 1]
```diff
@@ -57,6 +58,7 @@ NS_ASSUME_NONNULL_BEGIN
 - (void)setPictureInPicture:(BOOL)pictureInPicture;
 - (void)disablePictureInPicture;
 - (void)willStartPictureInPicture:(bool)willStart;
+- (void)setIsHidedPipSeekButtons:(BOOL) isHided;
```
Comment (mvn-thanhluu-hn) [inline ios/Classes/BetterPlayer.h]:
```suggestion
- (void)setPipSeekButtonsHidden:(BOOL) isHidden;
```

File: ios/Classes/BetterPlayerPlugin.m [hunk 2]
```diff
@@ -417,6 +417,10 @@ - (void)handleMethodCall:(FlutterMethodCall*)call result:(FlutterResult)result {
             } else {
                 [self disposeNotificationData:player];
             }
+            
+            BOOL isLiveStream = [self isLiveStream:player];
+            [player setIsHidedPipSeekButtons:isLiveStream];
+            player._isLiveStream = isLiveStream;
```
Comment (mvn-thanhluu-hn) [inline ios/Classes/BetterPlayerPlugin.m]: We use `_` ofter for property only use in parent class, please make a set function for it.

**Review by mvn-thanhluu-hn — COMMENTED** [review]
@mvn-hienvu-hn 
Sorry, please check my comments 🙏

**Review by mvn-thanhluu-hn — APPROVED** [review]
LGTM.

@mvn-hienvu-hn Thank you.

File: lib/src/configuration/better_player_data_source.dart [hunk 3]
```diff
@@ -158,6 +158,7 @@ class BetterPlayerDataSource {
     List<BetterPlayerSubtitlesSource>? subtitles,
     bool? useAsmsSubtitles,
     bool? useAsmsTracks,
+    bool? liveStream,
```
Comment (HiroshiOshiro) [inline lib/src/configuration/better_player_data_source.dart]: Please change to `isLiveStream` (in every part)

**Review by HiroshiOshiro — CHANGES_REQUESTED** [review]
@mvn-hienvu-hn 
I have a small request. please check it.
(Please update flutter repo too)

**Review by HiroshiOshiro — APPROVED** [review]
LGTM

**Pull request merged**